xwiki-platform-core/xwiki-platform-websocket/src/main/java/org/xwiki/websocket/script/WebSocketScriptService.java

```diff
@@ -84,9 +84,13 @@ public String url(String pathOrRoleHint)
             XWikiContext xcontext = this.xcontextProvider.get();
 
             StringBuilder path = new StringBuilder("/");
-            path.append(xcontext.getWiki().getWebAppPath(xcontext));
-            path.append("websocket").append(path(pathOrRoleHint));
 
+            // Test if installed with short URL and avoid double slash
+            if (!"/".equals(xcontext.getWiki().getWebAppPath(xcontext))) {
+                path.append(xcontext.getWiki().getWebAppPath(xcontext));
+            }
```

Comment:
Instead of this, why not simply change the line above and remove this IF:

From:

```
StringBuilder path = new StringBuilder("/");
```

To:

```
StringBuilder path = new StringBuilder();
```

?

Reply:
if `getWebAppPath()`  doesn't always return a path prefixed with `/` then just check for that if it doesn't start with `/` then add it.

Reply:
Actually just checked and indeed, `getWebAppPath()` never returns a leading "/", see https://github.com/xwiki/xwiki-platform/blob/67c0e919f2c3d0e4385b2133e7cc0318da598834/xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/XWiki.java#L5456

Reply:
It does not return a leading "/" but always a trailing one, right?

My idea was to exclude this single edge case where the leading "/" is equal to the trailing one which leads to this code returning "//". Or do I get you wrong?

Reply:
@promasu thanks, what you did is good.

Reply:
Ok perfect. Thanks!